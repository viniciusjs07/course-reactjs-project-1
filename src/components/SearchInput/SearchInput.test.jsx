import userEvent from '@testing-library/user-event';
import { SearchInput } from '.';
import { render, screen } from '@testing-library/react';
const fn = jest.fn();

describe('<SearchInput/>', () => {
  it('should have a value of searchValue', () => {
    render(<SearchInput handleSearch={fn} searchValue={'insert text'} />);
    const input = screen.getByPlaceholderText(/Pesquise seu post aqui/i);
    expect(input).toBeInTheDocument();
    expect(input.value).toBe('insert text');

    expect(screen.getByRole('searchbox')).toBeInTheDocument();
    expect.assertions(3);
  });
  it('should call handleChange function on each kry pressed', () => {
    render(<SearchInput handleSearch={fn} />);

    const input = screen.getByPlaceholderText(/Pesquise seu post aqui/i);

    const value = 'novo valor do input';
    userEvent.type(input, value);

    expect(input.value).toBe('novo valor do input');
    // check how many times the value was entered
    expect(fn).toHaveBeenCalledTimes(19);
    expect(screen.getByRole('searchbox')).toBeInTheDocument();
    expect.assertions(3);
  });

  it('Should match snapshot', () => {
    const { container } = render(<SearchInput handleSearch={fn} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
