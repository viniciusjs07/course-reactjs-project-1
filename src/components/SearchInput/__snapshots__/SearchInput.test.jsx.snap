// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput/> Should match snapshot 1`] = `
<input
  class="search-input"
  placeholder="Pesquise seu post aqui"
  type="search"
  value=""
/>
`;
